app/src/androidTest/java/com/github/polypoly/app/commons/PolyPolyTest.kt

```diff
@@ -0,0 +1,157 @@
+package com.github.polypoly.app.commons
+
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import com.github.polypoly.app.game.GameLobby
+import com.github.polypoly.app.game.GameMode
+import com.github.polypoly.app.game.user.Skin
+import com.github.polypoly.app.game.user.Stats
+import com.github.polypoly.app.game.user.User
+import com.github.polypoly.app.global.GlobalInstances
+import com.github.polypoly.app.global.Settings
+import com.github.polypoly.app.global.Settings.Companion.DB_GAME_LOBIES_PATH
+import com.github.polypoly.app.global.Settings.Companion.DB_USERS_PROFILES_PATH
+import com.github.polypoly.app.network.RemoteDB
+import com.google.firebase.database.DatabaseReference
+import com.google.firebase.database.ktx.database
+import com.google.firebase.ktx.Firebase
+import org.junit.Before
+import org.junit.runner.RunWith
+import java.util.concurrent.CompletableFuture
+import java.util.concurrent.TimeUnit
+import java.util.concurrent.locks.ReentrantLock
+import kotlin.time.Duration.Companion.hours
+import kotlin.time.DurationUnit
+
+@RunWith(AndroidJUnit4::class)
+abstract class PolyPolyTest(
+    private val clearRemoteStorage: Boolean, //> clear remote storage at the beginning of every test
+    private val fillWithFakeData: Boolean //> fill remote storage with fake data at the beginning of every test
+) {
```

Comment:
This is a bit confusing, first because they're Booleans so just reading the code it's not clear what, e.g., `PolyPolyTest(true, false)` means, but also why would a test want the `false, true` option? `JoinGameLobbyActivityTest` does this above but what does this mean for the test, since it presumably won't know what other data there is aside from the test data? An enum with three options "leave database unchanged", "clear database", and "set database to fake data" would be more readable and make more sense IMHO.